report/src/sections/04-theory/06-termination.tex

```diff
@@ -0,0 +1,20 @@
+\subsection{Termination}\label{subsec:termination}
+
+In this subsection we will argument for our analysis terminating.
+
+Our analysis takes base in the semantics for instructions $\abssem{I}$.
+Hence, to prove that our analysis terminates, we would need to show that $\abssem{I}$ always reaches a fixed point.
+
+As seen in \autoref{eq:abssemtype}, $\abssem{I}$ maps to $\mathcal{P}(\ab{\mathfrak{E}})$.
+We know from \autoref{thm:csql} that $\abssem{I}$ is a monotone function.
+So from \autoref{thm:kleene_finite}, we see that, in order for the analysis to reach a fixed point in a finite number of steps, $\mathcal{P}(\ab{\mathfrak{E}})$ has to be defined over a finite and complete lattice.
```

Comment:
The function I talked about above will be defined for tuples over the power set of abstract environments, the power set is a complete lattice and therefore the tuples defined over are also a complete lattice and the same for finiteness.